core/src/main/java/org/ldaptive/transport/DefaultCompareOperationHandle.java

```diff
@@ -146,6 +146,12 @@ public DefaultCompareOperationHandle onCompare(final CompareValueHandler... func
    */
   public void compare(final CompareResponse response)
   {
+    if (getMessageID() != response.getMessageID()) {
```

Comment:
I don't understand the need for this check. Is it needed for unsolicited responses?

Reply:
Understood, thanks for the explanation.